lib/btc/BtcRpc.js

```diff
@@ -53,65 +64,81 @@ class BtcRpc {
     if (passphrase === undefined) {
       passphrase = await passwordPromptAsync('> ');
     }
-    await this.walletUnlock({ passphrase, time: 10 });
+    await this.walletUnlock({ passphrase, time: 10800 });
     const tx = await this.sendToAddress({ address, amount });
     await this.walletLock();
     return tx;
   }
 
-  async unlockAndSendToAddressMany({ payToArray, passphrase, time = 10800 }) {
-    function RpcException(message) {
-      const e = new Error(message);
-      e.name = 'failedRequest';
-      e.data = errorObject;
-      return e;
-    }
-
+  async unlockAndSendToAddressMany({ payToArray, passphrase, time=10800, maxValue=10*1e8, maxOutputs=1 }) {
     if (passphrase === undefined) {
       passphrase = await passwordPromptAsync('> ');
     }
+    if (payToArray.length < maxOutputs) {
+      maxOutputs = payToArray.length;
+    }
     await this.walletUnlock({ passphrase, time });
-    this.emitter.emit('unlocked', time);
-    let someRequestFailed;
-    const errorObject = { success: {}, failure: {} };
-    const result = [];
-    for (const [i, request] of payToArray.entries()) {
-      const { address, amount } = request;
-      const emitData = {
-        address: request.address,
-        amount: request.amount,
-        id: request.id
+    let payToArrayResult = [];
+    while(payToArray.length) {
+      let currentValue = 0;
+      let currentOutputs = 0;
+      let batch = {};
+      let tempArray = [];
+      while(currentValue < maxValue && currentOutputs < maxOutputs) {
+        let tx = payToArray.shift();
```

Comment:
Does this have side-effects?

Reply:
This only returns and removes the first element, no side-effects that I am aware of

Reply:
If I pass an array into this method, does this mutate my array?

Reply:
you pass in an array of objects and get back the same array of objects but with additional params in each object